orchagent/vxlanorch.cpp

```diff
@@ -349,16 +349,13 @@ create_tunnel(
         tunnel_attrs.push_back(attr);
     }
 
-    if (encap_ttl != 0)
-    {
-        attr.id = SAI_TUNNEL_ATTR_ENCAP_TTL_MODE;
-        attr.value.s32 = SAI_TUNNEL_TTL_MODE_PIPE_MODEL;
-        tunnel_attrs.push_back(attr);
+    attr.id = SAI_TUNNEL_ATTR_ENCAP_TTL_MODE;
```

Comment:
The current behaviour of this helper function allowed for uniform as well as pipe model with the default being uniform (encap_ttl=0) 
With this change sonic forces tunnels to be in pipe mode which is a step back. 

Hence I would say stick with the current implementation.

Reply:
What's the solution here to the issue at hand then?  What we're seeing is ARP across the tunnel no longer works as it gets a TTL of 0 and gets dropped, therefore machines are unable to find eachother across the network.  At least on Broadcom Trident 3.

Reply:
or are you saying to try to just set a TTL > 0 in Uniform mode?  So just don't change the mode attribute and see what happens?

Reply:
Is the ARP TTL issue on Trident3 still seen with the latest SAI versions ?

Reply:
as of 3 weeks ago on master anyhow, I haven't tried without this patch since then

Reply:
@srj102 how do you want me to proceed here?  I can test anything you want me to test within a few hours, such as setting the TTL be leaving in uniform mode.  That said, I must say with VXLAN EVPN I'm not sure uniform mode make sense anyhow unless I'm missing something.  MPLS it does of course.  I've also seen some vendors indicate they *only* support pipe mode on VXLAN EVPN (https://documentation.extremenetworks.com/slxos/sw/20xx/20.4.3/l2config/GUID-537EC20E-95A2-4CA7-A417-D45D596AD575.shtml)

Reply:
The default of uniform mode seems to be coming from SAI defaults.

The foll is my suggestion. 
a. Not to change the create_tunnel helper function. 
b. Pass a non default encap_ttl from createTunnelHw()

I am fine with having the default encap ttl mode to be PIPE. 
@prsunny  @dgsudharsan  please comment. 

If this is not acceptable,
 then we might have to have this as PIPE only for BRCM platforms. 
Define a default_encap_ttl and set it to PIPE only for BRCM platforms (platform == BRCM_PLATFORM_SUBSTRING )

Reply:
@srj102 ping

Reply:
@srj102 I was really just asking for feedback on the latest changeset, if that's more inline with what you were wanting...